Library/Homebrew/cmd/update.sh

```diff
@@ -781,68 +781,73 @@ EOS
       then
-        JSON_URLS=("${HOMEBREW_API_DOMAIN}/${formula_or_cask}.jws.json")
+        JSON_URLS=("${HOMEBREW_API_DOMAIN}/${json}.json")
       fi
 
-      JSON_URLS+=("${HOMEBREW_API_DEFAULT_DOMAIN}/${formula_or_cask}.jws.json")
+      JSON_URLS+=("${HOMEBREW_API_DEFAULT_DOMAIN}/${json}.json")
       for json_url in "${JSON_URLS[@]}"
       do
         time_cond=()
-        if [[ -s "${HOMEBREW_CACHE}/api/${formula_or_cask}.jws.json" ]]
+        if [[ -s "${cache_path}" ]]
         then
-          time_cond=("--time-cond" "${HOMEBREW_CACHE}/api/${formula_or_cask}.jws.json")
+          time_cond=("--time-cond" "${cache_path}")
         fi
         curl \
           "${CURL_DISABLE_CURLRC_ARGS[@]}" \
           --fail --compressed --silent \
           --speed-limit "${HOMEBREW_CURL_SPEED_LIMIT}" --speed-time "${HOMEBREW_CURL_SPEED_TIME}" \
-          --location --remote-time --output "${HOMEBREW_CACHE}/api/${formula_or_cask}.jws.json" \
+          --location --remote-time --output "${cache_path}" \
           "${time_cond[@]}" \
           --user-agent "${HOMEBREW_USER_AGENT_CURL}" \
           "${json_url}"
         curl_exit_code=$?
         [[ ${curl_exit_code} -eq 0 ]] && break
       done
 
-      if [[ -f "${HOMEBREW_CACHE}/api/${formula_or_cask}_names.txt" ]]
+      if [[ "${json}" == "formula.jws" ]] && [[ -f "${api_cache}/formula_names.txt" ]]
       then
-        mv -f "${HOMEBREW_CACHE}/api/${formula_or_cask}_names.txt" \
-          "${HOMEBREW_CACHE}/api/${formula_or_cask}_names.before.txt"
+        mv -f "${api_cache}/formula_names.txt" "${api_cache}/formula_names.before.txt"
+      elif [[ "${json}" == "cask.jws" ]] && [[ -f "${api_cache}/cask_names.txt" ]]
+      then
+        mv -f "${api_cache}/cask_names.txt" "${api_cache}/cask_names.before.txt"
       fi
 
       if [[ ${curl_exit_code} -eq 0 ]]
       then
-        touch "${HOMEBREW_CACHE}/api/${formula_or_cask}.jws.json"
+        touch "${cache_path}"
 
-        CURRENT_JSON_BYTESIZE="$(wc -c "${HOMEBREW_CACHE}"/api/"${formula_or_cask}".jws.json)"
+        CURRENT_JSON_BYTESIZE="$(wc -c "${cache_path}")"
         if [[ "${INITIAL_JSON_BYTESIZE}" != "${CURRENT_JSON_BYTESIZE}" ]]
         then
 
-          if [[ "${formula_or_cask}" == "formula" ]]
+          if [[ "${json}" == "formula.jws" ]]
           then
-            rm -f "${HOMEBREW_CACHE}/api/formula_aliases.txt"
+            rm -f "${api_cache}/formula_aliases.txt"
           fi
           HOMEBREW_UPDATED="1"
         fi
       else
         echo "Failed to download ${json_url}!" >>"${update_failed_file}"
       fi
 
-      # Not a typo, this is the file we used to download that we should cleanup.
-      rm -f "${HOMEBREW_CACHE}/api/${formula_or_cask}.json"
     done
+
+    # Not a typo, these are the files we used to download that no longer need so should cleanup.
+    rm -f "${HOMEBREW_CACHE}/api/formula.json" "${HOMEBREW_CACHE}/api/cask.json"
```

Comment:
The `-f` means declaring these explicitly and doing a single call is a bit nicer.